network/connection.go

```diff
@@ -19,48 +18,114 @@ import (
+	// created from a DNS request.
 	IPProtocol packet.IPProtocol
-	LocalIP    net.IP
-	LocalPort  uint16
-	process    *process.Process
-
-	// remote endpoint
+	// LocalIP holds the local IP address of the connection. It is not
+	// set for connections created from DNS requests. LocalIP is
+	// considered immutable once a connection object has been created.
+	LocalIP net.IP
+	// LocalPort holds the local port of the connection. It is not
+	// set for connections created from DNS requests. LocalPort is
+	// considered immutable once a connection object has been created.
+	LocalPort uint16
+	// Entity describes the remote entity that the connection has been
+	// established to. The entity might be changed or information might
+	// be added to it during the livetime of a connection. Access to
+	// entity must be guarded by the connection lock.
 	Entity *intel.Entity
-
-	Verdict       Verdict
-	Reason        string
+	// Verdict is the final decision that has been made for a connection.
+	// The verdict may change so any access to it must be guarded by the
+	// connection lock.
+	Verdict Verdict
+	// Reason is a human readable description justifying the set verdict.
+	// Access to Reason must be guarded by the connection lock.
+	Reason string
+	// ReasonContext may holds additional reason-specific information and
+	// any access must be guarded by the connection lock.
 	ReasonContext interface{}
-	ReasonID      string // format source[:id[:id]] // TODO
-
-	Started          int64
-	Ended            int64
-	Tunneled         bool
+	// Started holds the number of seconds in UNIX epoch time at which
+	// the connection has been initated and first seen by the portmaster.
+	// Staretd is only every set when creating a new connection object
+	// and is considered immutable afterwards.
+	Started int64
+	// Ended is set to the number of seconds in UNIX epoch time at which
+	// the connection is considered terminated. Ended may be set at any
+	// time so access must be guarded by the conneciton lock.
+	Ended int64
+	// VerdictPermanent is set to true if the final verdict is permanent
+	// and the connection has been (or will be) handed back to the kernel.
+	// VerdictPermanent may be changed together with the Verdict and Reason
+	// properties and must be guarded using the connection lock.
 	VerdictPermanent bool
-	Inspecting       bool
-	Encrypted        bool // TODO
-	Internal         bool // Portmaster internal connections are marked in order to easily filter these out in the UI
-
-	pktQueue        chan packet.Packet
+	// Inspecting is set to true if the connection is being inspected
+	// by one or more of the registered inspectors. This property may
+	// be changed during the lifetime of a connection and must be guarded
+	// using the connection lock.
+	Inspecting bool
+	// Tunneled is currently unused and MUST be ignored.
+	Tunneled bool
+	// Encrypted is currently unused and MUST be ignored.
+	Encrypted bool
```

Comment:
Both of these are currently being worked on. Should we not add a description already?